src/litserve/api.py

```diff
@@ -68,5 +68,9 @@ def unbatch(self, output):
 
     @abstractmethod
     def encode_response(self, output):
-        """Convert the model output to a response payload."""
+        """Convert the model output to a response payload.
+
+        To enable streaming, it should yield the output.
```

Comment:
in `LitServer` we should check that these functions are actually both yielding using
`inspect.isgeneratorfunction`
when `streaming=True`, and they are not we should return a nice error message

Reply:
Also please add tests that test this behavior.

Reply:
added the check and test - https://github.com/Lightning-AI/litserve/pull/37/files#diff-274978086d8b4f049cffd0a53c0a2b1634c323f7ae581e87427c3b0c9046716aR174